server/middleware/github-griller/github.parse.js

```diff
@@ -0,0 +1,106 @@
+function repo(body) {
+  if(!body) return null;
+  var repo = {};
+  repo.githubId = body.id || null;
+  repo.name = body.name || null;
+  repo.description = body.description || null;
+  //Use presave hook to turn this field into a proper User
+  repo.owner = {
+    username: body.owner.login,
+    githubId: body.owner.id,
+    url: body.owner.url
+  };
+
+  repo.url = body.url || null;
+  repo.collaborators_url = body.collaborators_url || null;
+
+  return repo;
+}
+
+function repos(body) {
+  return body.map(function(item) {
+    return repo(item)
+  });
+}
+
+function issue(body) {
+  if(!body) return null;
+  var issue = {};
+  issue.githubId = body.id;
+  issue.issueNumber = 0 + body.number;
+  issue.title = body.title;
+  issue.body = body.body;
+  issue.state = body.state;
+  issue.milestone = body.milestone;
+  issue.labels = body.labels;
+  issue.created_at = body.created_at;
+  issue.updated_at = body.updated_at;
+  issue.closed_at = body.closed_at;
+  issue.due_on = body.due_on;
+  issue.url = body.url;
+  issue.labels_url = body.labels_url;
+  issue.events_url = body.events_url;
+  issue.html_url = body.html_url;
+
+  issue.owner = {
+    username: body.user.login,
+    githubId: body.user.id,
+    url: body.user.url
+  };
+
+  if (body.assignee) issue.assignee =  body.assignee.login;
+  if (body.pull_request) issue.isPullRequest = true;
+
+  return issue;
+}
+
+function issues(body) {
+  return body.map(function(item) {
+    return issue(item)
+  });
+}
+
+// This function may need to do some clever work to figure out which issue a comment belongs to.
+function comment(body) {
+  if(!body) return null;
+  var comment = {};
+  comment.githubId = body.id;
+  comment.body = body.body;
+  comment.path = body.path;
+  comment.position = body.position;
+  comment.line = body.line;
+  comment.commit_id = body.commit_id;
+  comment.user = body.user;
+  comment.created_at = body.created_at;
+  comment.updated_at = body.updated_at;
+
+  return body;
+}
+
+function comments(body) {
+  return body.map(function(item) {
+    return comment(item)
+  });
+}
+
+function parse(req) {
```

Comment:
@ssarda77 Not totally sure how to make use of the parsing.  We could use this function 'parse' as a sort of middleware - you give it a req, and it parses everything of importance on the req before passing it on to the next middleware.  Alternatively, we could use these different functions that are exported selectively whenever we need them to parse particular things.